### ISSUE TITLE
degradation loss

### ISSUE THREAD

**Issue by sucessfullys** [XunpengYi/Text-IF#21]
Hello, thank you very much for your work. When I perform image fusion under rainy degradation, the fused images often exhibit abnormal colors. Could you please explain what might be causing this issue? Also, could you provide more details about the experimental setup for rain degradation?

**Comment by XunpengYi** [comment]
We do not encounter significant color differences during the deraining task. Could you please provide more detailed information?

**Comment by sucessfullys** [comment]
Thank you very much for your reply. I used the code you provided to
generate rain-degraded visible images and normal infrared images, and fed
them into the network for training. Since the code does not provide
degradation ratios for rain, blur, and haze, we used the fusion_loss
function with the following ratios: max_ratio=4, ssim_vis_ratio=1,
ssim_ir_ratio=1, ssim_ratio=1, color_ratio=12, text_ratio=10. The ground
truth images are normal visible and infrared images. However, the fused
results show significant color deviations. Could you please provide
degradation loss functions related to rain, blur, and haze? I would greatly
appreciate your help.

Xunpeng Yi ***@***.***> 于2025年5月30日周五 10:49写道：

**Comment by sucessfullys** [comment]
> We do not encounter significant color differences during the deraining task. Could you please provide more detailed information?
When I incorporated loss functions related to rain, blur, and haze into the training process, using the EMS dataset and keeping all other training parameters unchanged, some fused output images exhibited noticeable purple and green artifacts. The loss functions were added as follows, with all other settings remaining the same as in the original code. I suspect the issue may be due to improper weighting of the loss functions, as the author did not provide complete loss configurations for all tasks. I hope to get clarification on this.
for idx, task_type in enumerate(task):
            img_A = self._get_image(image_A, idx)
            img_B = self._get_image(image_B, idx)
            img_fused = self._get_image(image_fused, idx)

            if task_type == "low_light":
                loss, ssim_loss, max_loss, color_loss, grad_loss = self.fusion_loss(img_A, img_B, img_fused,
                                                                                    max_ratio=8, ssim_ratio=1, text_ratio=10)
            elif task_type == "over_exposure":
                loss, ssim_loss, max_loss, color_loss, grad_loss = self.fusion_loss(img_A, img_B, img_fused,
                                                                                    max_ratio=4, ssim_ratio=0, text_ratio=2)
            elif task_type == "rain":
                loss, ssim_loss, max_loss, color_loss, grad_loss = self.fusion_loss(img_A, img_B, img_fused,
                                                                                    max_ratio=8, ssim_ratio=1, text_ratio=10)
            elif task_type == "random_noise":
                loss, ssim_loss, max_loss, color_loss, grad_loss = self.fusion_loss(img_A, img_B, img_fused,
                                                                                    max_ratio=6, ssim_ratio=1, text_ratio=10)
            elif task_type == "haze":
                loss, ssim_loss, max_loss, color_loss, grad_loss = self.fusion_loss(img_A, img_B, img_fused,
                                                                                    max_ratio=8, ssim_ratio=1, text_ratio=10)
            elif task_type == "blur":
                loss, ssim_loss, max_loss, color_loss, grad_loss = self.fusion_loss(img_A, img_B, img_fused, max_ratio=8, ssim_ratio=1, text_ratio=10)
            else:
                raise ValueError(f"Unknown task type: {task_type}")

Thanks very much!

**Comment by XunpengYi** [comment]
It seems that this might be related to the magnitude of the color loss. You could try adjusting this coefficient. Could you provide the inference results obtained from the training?

**Comment by sucessfullys** [comment]
> It seems that this might be related to the magnitude of the color loss. You could try adjusting this coefficient. Could you provide the inference results obtained from the training?

like this
![Image](https://github.com/user-attachments/assets/ae39c911-5400-434c-9f7d-36a35c738e75)

Could you please supplement the loss weight settings for the other tasks in the GitHub code? Thanks very much

**Comment by XunpengYi** [comment]
For the deraining task, we recommend setting max_ratio = 6, ssim_ratio = 1, color_ratio = 12, and text_ratio = 10. You can give it a try.

**Comment by sucessfullys** [comment]
> For the deraining task, we recommend setting max_ratio = 6, ssim_ratio = 1, color_ratio = 12, and text_ratio = 10. You can give it a try.

Thanks very much. What are the loss settings for blur, haze, and vis_noise? I'm trying to perform mixed training for reproduction and comparison. Thank you very much!

**Comment by sucessfullys** [comment]
> For the deraining task, we recommend setting max_ratio = 6, ssim_ratio = 1, color_ratio = 12, and text_ratio = 10. You can give it a try.

Looking forward to your reply. Thank you very much.

**Comment by sucessfullys** [comment]
> For the deraining task, we recommend setting max_ratio = 6, ssim_ratio = 1, color_ratio = 12, and text_ratio = 10. You can give it a try.
For the deraining task, I followed these settings: max_ratio = 6, ssim_ratio = 1, color_ratio = 12, and text_ratio = 10. The loss weights for low-light, IR noise, and over-exposure were also set according to the original code. However, for the "more degradation (all types in EMS)" task, there are multiple possible configurations for loss function ratios (e.g., for blur, haze, etc.), making it difficult to combine them effectively. Preliminary experiments also encountered color abnormalities.

Therefore, I hope the authors could supplement the complete loss function settings for all degradation types in EMS on GitHub. I would greatly appreciate your response and thank you again for your work!
